format/luajit/luajit.go

```diff
@@ -0,0 +1,381 @@
+package luajit
+
+// dump   = header proto+ 0U
+// header = ESC 'L' 'J' versionB flagsU [namelenU nameB*]
+// proto  = lengthU pdata
+// pdata  = phead bcinsW* uvdataH* kgc* knum* [debugB*]
+// phead  = flagsB numparamsB framesizeB numuvB numkgcU numknU numbcU
+//          [debuglenU [firstlineU numlineU]]
+// kgc    = kgctypeU { ktab | (loU hiU) | (rloU rhiU iloU ihiU) | strB* }
+// knum   = intU0 | (loU1 hiU)
+// ktab   = narrayU nhashU karray* khash*
+// karray = ktabk
+// khash  = ktabk ktabk
+// ktabk  = ktabtypeU { intU | (loU hiU) | strB* }
+//
+// B = 8 bit, H = 16 bit, W = 32 bit, U = ULEB128 of W, U0/U1 = ULEB128 of W+1
+
+// see:
+//
+//	* http://scm.zoomquiet.top/data/20131216145900/index.html
+//	* https://github.com/LuaJIT/LuaJIT/blob/v2.1/src/lj_bcdump.h
+
+import (
+	"bytes"
+	"embed"
+	"encoding/binary"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+//go:embed luajit.md
+var LuaJITFS embed.FS
+
+var probeGroup decode.Group
+
+func init() {
+	interp.RegisterFormat(
+		format.LuaJIT,
+		&decode.Format{
+			Description: "LuaJIT 2.0 bytecode",
+			Groups:      []*decode.Group{format.Probe},
+			DecodeFn:    LuaJITDecode,
+			Dependencies: []decode.Dependency{
```

Comment:
`Dependencies` you can skip, it's if the format decoder itself want to decode something using the probe group